src/Symfony/Component/TypeInfo/TypeResolver/StringTypeResolver.php

```diff
@@ -119,6 +120,40 @@ private function getTypeFromNode(TypeNode $node, ?TypeContext $typeContext): Typ
         }
 
         if ($node instanceof ConstTypeNode) {
+            if ($node->constExpr instanceof ConstFetchNode) {
+                $types = [];
+
+                $className = match (strtolower($node->constExpr->className)) {
+                    'self' => $typeContext->getDeclaringClass(),
+                    'static' => $typeContext->getCalledClass(),
+                    'parent' => $typeContext->getParentClass(),
+                    default => $node->constExpr->className,
+                };
+
+                foreach ((new \ReflectionClass($className))->getReflectionConstants() as $const) {
```

Comment:
this should probably handle the case where the class does not exist, instead of getting a ReflectionException (I would suggest to use `mixed` as fallback when we cannot find the referenced class)